From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Indhi Rousseau <contact@thekinrar.fr>
Date: Sun, 27 Oct 2019 02:19:43 +0000
Subject: [PATCH] Add PluginDescriptionFile getter to PluginClassLoader

Can be used to get plugin info before main class has been loaded

diff --git a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
index 7760be3e34fa20825faf145d9fb5b2855c1a4602..6f99f08cbba41d76f6aa6d4d5a09b0f2b77782a3 100644
--- a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
+++ b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
@@ -30,6 +30,7 @@ import org.jetbrains.annotations.Nullable;
  */
 public final class PluginClassLoader extends URLClassLoader { // Spigot
     public JavaPlugin getPlugin() { return plugin; } // Spigot
+    public PluginDescriptionFile getDescription() { return description; } // Papyrus
     private final JavaPluginLoader loader;
     private final Map<String, Class<?>> classes = new ConcurrentHashMap<String, Class<?>>();
     private final PluginDescriptionFile description;
